src/ehrdata/tl/pandas/_to_pandas.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Literal
+
+import pandas as pd
+import xarray as xr
+from scipy.sparse import issparse
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from ehrdata import EHRData
+
+
+def to_pandas(
+    edata: EHRData,
+    layer: str | None = None,
+    obs_cols: Iterable[str] | str | None = None,
+    var_col: str | None = None,
+    format: Literal["wide", "long"] = "wide",
+) -> pd.DataFrame:
+    """Transform an EHRData object to a Pandas DataFrame.
+
+    Args:
+        edata: The EHRData object to be transformed into a pandas DataFrame
+        layer: The layer to access the values of. If not specified, it uses the `X` matrix.
+        obs_cols: The columns of `obs` to add to the DataFrame.
+        var_col: The column of `var` to create the column names from in the created DataFrame. If not specified, the `var_names` will be used.
+        format: The format of the output DataFrame. This is relevant for longitudinal data. If "wide", the output dataframe will write a column for each (variable, time) tuple, naming the colun as <variable_name>_t_<tem.index value>. If "long", the output dataframe will be in long format, with columns "observation_id", "variable", "time", and "value".
+
+    Returns:
+        The data object as a pandas DataFrame.
+
+    Examples:
+        >>> import ehrdata as ed
+        >>> edata = ed.dt.mimic_2()
+        >>> df = ep.tl.to_pandas(edata)
+    """
+    X = edata.layers[layer] if layer is not None else edata.X
+
+    if var_col is not None and var_col not in edata.var.columns:
+        err_msg = f"Variable column {var_col} not found in edata.var"
+        raise ValueError(err_msg)
+
+    var_names = edata.var_names if var_col is None else edata.var[var_col]
+
+    if issparse(X):  # pragma: no cover
+        X = X.toarray()
```

Comment:
Use fast-array-utils.

IDK whether we want to depend on it here but maybe the non numba version (without accel extra)